test/CompanyService.Business.UnitTests/Commands/Company/CreateCompanyCommandTests.cs

```diff
@@ -1,344 +1,544 @@
+                .Returns(true);
+        }
+
+        [Test]
+        public void ShouldThrowValidationException()
+        {
+            _autoMock
+               .Setup<ICreateCompanyRequestValidator, bool>(
+               x => x.Validate(It.IsAny<IValidationContext>()).IsValid)
+               .Returns(false);
+
+            Assert.Throws<ValidationException>(() => _command.Execute(_request));
+
+            _autoMock.Verify<ICompanyRepository>(
+               x => x.Add(It.IsAny<DbCompany>()),
+               Times.Never());
+
+            _autoMock.Verify<ICreateCompanyRequestValidator>(
+               x => x.Validate(It.IsAny<IValidationContext>()),
+               Times.Once);
+
+            _autoMock.Verify<IDbCompanyMapper>(
+               x => x.Map(It.IsAny<CreateCompanyRequest>()),
+               Times.Never());
+
+            _autoMock.Verify<IRequestClient<IUpdateSmtpCredentialsRequest>>(
+               x => x.GetResponse<IOperationResult<bool>>(
+            It.IsAny<object>(), default, default).Result.Message, Times.Never);
+
+            _autoMock.Verify<IRequestClient<ICreateAdminRequest>>(
+               x => x.GetResponse<IOperationResult<bool>>(
+               It.IsAny<object>(), default, default).Result.Message, Times.Never);
+        }
+
+        [Test]
+        public void ShouldThrowBadRequestException()
+        {
+            _autoMock
+               .Setup<ICompanyRepository, DbCompany>(x => x.Get(null))
+               .Returns(_company);
+
+            var ex = Assert.Throws<BadRequestException>(() => _command.Execute(_request));
+            Assert.That(ex.Message, Is.EqualTo("Company already exists"));
+
+            _autoMock.Verify<ICompanyRepository>(
+               x => x.Add(It.IsAny<DbCompany>()),
+               Times.Never());
+
+            _autoMock.Verify<ICreateCompanyRequestValidator>(
+               x => x.Validate(It.IsAny<IValidationContext>()),
+               Times.Never);
+
+            _autoMock.Verify<IDbCompanyMapper>(
+               x => x.Map(It.IsAny<CreateCompanyRequest>()),
+               Times.Never());
+
+            _autoMock.Verify<IRequestClient<IUpdateSmtpCredentialsRequest>>(
+               x => x.GetResponse<IOperationResult<bool>>(
+            It.IsAny<object>(), default, default).Result.Message, Times.Never);
+
+            _autoMock.Verify<IRequestClient<ICreateAdminRequest>>(
+               x => x.GetResponse<IOperationResult<bool>>(
+               It.IsAny<object>(), default, default).Result.Message, Times.Never);
+        }
+
+        [Test]
+        public void ShouldReturnFailedResponseWhenUpdateSMTPResponseThrowAndCatchException()
+        {
+            _autoMock
+                .Setup<IOperationResult<bool>, bool>(x => x.IsSuccess)
+                .Returns(false);
+
+            _autoMock
+                .Setup<IOperationResult<bool>, List<string>>(x => x.Errors)
+                .Returns(new List<string> { "some error" });
```

Comment:
"Can not update smtp credentials." - the error message

Reply:
done